tests/unit/api/test_deepset_cloud_api.py

```diff
@@ -66,6 +66,31 @@ async def test_get(
             timeout=123,
         )
 
+    async def test_get_retry(
```

Comment:
we should also add the cases for:

- all errors
- first error is not of retryable category (e.g. httpx.ConnectError)

Reply:
I added a retry for all `RequestError`'s

Reply:
I added the parent class `RequestError` lets just test for this and one child.  This should be enough, since we would otherwise test httpx intead of our code

Reply:
done